WordPress/src/main/java/org/wordpress/android/ui/themes/ThemeBrowserActivity.java

```diff
@@ -234,100 +254,187 @@ public void onResponse(JSONObject response) {
+        fragmentTransaction.commit();
+    }
 
-        if (!DisplayUtils.isXLarge(ThemeBrowserActivity.this)) {
-            // show details as a fragment on top
-            FragmentTransaction ft = fm.beginTransaction();
+    private void addSearchFragment() {
+        if (mThemeSearchFragment == null) {
+            mThemeSearchFragment = new ThemeSearchFragment();
+        }
+        showSearchToolbar();
+        FragmentTransaction fragmentTransaction = getFragmentManager().beginTransaction();
+        fragmentTransaction.replace(R.id.theme_browser_container, mThemeSearchFragment);
+        fragmentTransaction.addToBackStack(null);
+        fragmentTransaction.commit();
+    }
 
-            if (mSearchFragment != null && mSearchFragment.isVisible()) {
-                fm.popBackStack();
+    private void activateTheme(String themeId) {
+        final String siteId = getBlogId();
+        final String newThemeId = themeId;
+
+        WordPress.getRestClientUtils().setTheme(siteId, themeId, new Listener() {
+            @Override
+            public void onResponse(JSONObject response) {
+                WordPress.wpDB.setCurrentTheme(siteId, newThemeId);
+                Theme newTheme = WordPress.wpDB.getTheme(siteId, newThemeId);
+                showAlertDialogOnNewSettingNewTheme(newTheme);
+                fetchCurrentTheme();
             }
+        }, new ErrorListener() {
+            @Override
+            public void onErrorResponse(VolleyError error) {
+                Toast.makeText(getApplicationContext(), R.string.theme_activation_error, Toast.LENGTH_SHORT).show();
+            }
+        });
+    }
 
-            setupBaseLayout();
-            mDetailsFragment = ThemeDetailsFragment.newInstance(themeId);
-            ft.add(R.id.theme_browser_container, mDetailsFragment, ThemeDetailsFragment.TAG);
-            ft.setTransition(FragmentTransaction.TRANSIT_FRAGMENT_FADE);
-            ft.addToBackStack(null);
-            ft.commit();
-        } else {
-            // show details as a dialog
-            mDetailsFragment = ThemeDetailsFragment.newInstance(themeId);
-            mDetailsFragment.show(getFragmentManager(), ThemeDetailsFragment.TAG);
-            getFragmentManager().executePendingTransactions();
-            int minWidth = getResources().getDimensionPixelSize(R.dimen.theme_details_dialog_min_width);
-            int height = getResources().getDimensionPixelSize(R.dimen.theme_details_dialog_height);
-            int width = Math.max((int) (DisplayUtils.getDisplayPixelWidth(this) * 0.6), minWidth);
-            mDetailsFragment.getDialog().getWindow().setLayout(width, height);
+    private void showAlertDialogOnNewSettingNewTheme(Theme newTheme) {
+        AlertDialog.Builder dialogBuilder = new AlertDialog.Builder(this);
+        dialogBuilder.setMessage(String.format(getString(R.string.theme_prompt), newTheme.getName(), newTheme.getAuthor()));
```

Comment:
BTW, I'm not a big fan of that Pop-up and the "Thanks for choosing". IMHO, a snack bar with an undo button would be better. It's no a big deal since users won't change theme every 5 minutes, keep it like this if you want to stay consistent with the Web.